### PR TITLE
Frontend spotify

### PR DESCRIPTION
**Pull request by ArgoNavis1763**
Spotify, ticketmaster and maps set up, tests working

### CODE REVIEW

File: cypress/e2e/auddio-API-tests.cy.js [hunk 1]
```diff
@@ -126,21 +137,21 @@ describe('API SUCCESS', () => {
...
 
 describe('API FAILURE', () => {
   it('should return an error message if the API call is unsuccessful', () => {
+    mockUser()
     cy.clock()
     cy.intercept('POST', 'http://localhost:3001/audio_info', {
       statusCode: 500,
     }).as('fetchAudioData')
-    cy.setCookie('accessToken', 'BQC5CHoBV44kZ6UHqNqG-xr_ZCF5kSjyjMWLLg49DXjaefrdkJjAXcd9bIixal716tuTahvywvwSIxYnH0bkSE7BDli9c57DcSco2iL2E3wBuHbAoniODOaLOs-K5YqDMq9EFIKow6KIG_Uv1NgYG6Lf2UogNv14YLanQCeSCXoE_XcFdGo2YYCEFjYg9Xy0Qvexelo')
+    cy.setCookie(
```
Comment (TomGroombridge) [inline cypress/e2e/auddio-API-tests.cy.js]: is it worth having a function to set the cookie as we do this alot?

File: cypress/e2e/spotify-ticketmaster-API-tests.cy.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    it('should show artist info when get more info button is clicked', () => {
+      cy.get('[data-automation="extra-info-card"]').contains('Top Tracks')
+    })
+  })
+})
+
+describe('artist with no upcoming shows', () => {
+  it.only('should return no upcoming shows if events is undefined', () => {
+    mockUser()
+    cy.intercept('POST', 'http://localhost:3001/audio_info', (req) => {
```
Comment (TomGroombridge) [inline cypress/e2e/spotify-ticketmaster-API-tests.cy.js]: could it be worth creating a function for this and maybe passing in the data we want 

```
MockAudioInfo('Fascination', 'Alphabeat')
```

File: cypress/support/commands.js [hunk 3]
```diff
@@ -1,25 +1,1504 @@
...
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
\ No newline at end of file
+export const mockUser = () => {
+  cy.intercept('GET', 'https://api.spotify.com/v1/me', (req) => {
+    req.reply({
+      country: 'GB',
```
Comment (TomGroombridge) [inline cypress/support/commands.js]: as there is so much data in this file could we maybe create a `constants` file in the cypress folder and hold the data seperately in there

For example.... 

```
const SPOTIFY_ME_DATA = {
      country: 'GB',
      display_name: 'Charlotte Elizabeth',
      email: 'charlielizzy@123.com',
      explicit_content: {
        filter_enabled: false,
        filter_locked: false,
      },
      external_urls: {
        spotify: 'https://open.spotify.com/user/1111882414',
      },
      followers: {
        href: null,
        total: 13,
      },
      href: 'https://api.spotify.com/v1/users/1111882414',
      id: '1111882414',
      images: [
        {
          height: null,
          url: 'https://i.scdn.co/image/ab6775700000ee8564d8255e4bcf6564cfe880d2',
          width: null,
        },
      ],
      product: 'premium',
      type: 'user',
      uri: 'spotify:user:1111882414',
    }
```

File: src/Pages/Home/index.jsx [hunk 4]
```diff
@@ -1,101 +1,94 @@
...
+  const [spotifyTrackID, setSpotifyTrackID] = useState('')
 
-const handleRemoveCookie = () => {
-removeCookie('accessToken', {
-  path: "/"
-  
-})
-navigate('/')
-}
+  // const handleRemoveCookie = () => {
```
Comment (TomGroombridge) [inline src/Pages/Home/index.jsx]: let's get rid of old code

File: src/Pages/Home/index.jsx [hunk 5]
```diff
@@ -1,101 +1,94 @@
...
+        <div className="m-5 p-3 bg-gray-900 text-align rounded-lg text-center w-60 text-3xl text-white">
+          <RecordingButton
+            setTitle={setTitle}
+            setArtist={setArtist}
+            setSongNotFound={setSongNotFound}
+            setCardState={setCardState}
+            setAPIError={setAPIError}
+            setLoading={setLoading}
+            loading={loading}
+            // setSpotifyArtistID={setSpotifyArtistID}
```
Comment (TomGroombridge) [inline src/Pages/Home/index.jsx]: remove

File: src/Pages/Home/index.jsx [hunk 6]
```diff
@@ -1,101 +1,94 @@
...
-            title={title}
-            album={album}
-            artist={artist}
-            artwork={artwork}
-          />
+          <SongCard title={title} artist={artist} />
         ) : null}
 
-        {title !== '' && cardState === 'artistCard' ? (
+        {/* {title !== '' && cardState === 'artistCard' ? (
```
Comment (TomGroombridge) [inline src/Pages/Home/index.jsx]: remove

File: src/Pages/Spotify/index.jsx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        name: data.artists[4].name,
+        url: data.artists[4].external_urls.spotify,
+      },
+    ])
+  }
+
+  const getArtistEvents = async (artistName) => {
+    const results = await fetch(
+      // `https://app.ticketmaster.com/discovery/v2/events?apikey=${process.env.REACT_APP_TICKETMASTER_API_KEY}&keyword=${artistName}&locale=*`
+      `https://app.ticketmaster.com/discovery/v2/events?apikey=0y3bmD7nU4GSr7rlKAPuFHbw48SeAdYn&keyword=${artistName}&locale=*`
```
Comment (TomGroombridge) [inline src/Pages/Spotify/index.jsx]: API key should be a ENV variable

File: src/Pages/Spotify/index.jsx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    }
+  }
+
+  // if (data !== null && topTracks !== null && relatedArtists !== null) {
+  //   setLoading(false)
+  // }
+
+  return (
+    <div>
+      {/* {data && topTracks && relatedArtists && events ? (
```
Comment (TomGroombridge) [inline src/Pages/Spotify/index.jsx]: remove old code

Comment (ArgoNavis1763) [reply src/Pages/Spotify/index.jsx]: done

File: src/Pages/Home/index.jsx (hunk 6 above)
Comment (ArgoNavis1763) [reply src/Pages/Home/index.jsx]: done

File: src/Pages/Home/index.jsx (hunk 5 above)
Comment (ArgoNavis1763) [reply src/Pages/Home/index.jsx]: done

File: src/Pages/Home/index.jsx (hunk 4 above)
Comment (ArgoNavis1763) [reply src/Pages/Home/index.jsx]: done

File: cypress/e2e/auddio-API-tests.cy.js (hunk 1 above)
Comment (ArgoNavis1763) [reply cypress/e2e/auddio-API-tests.cy.js]: setAccessToken function created and tests passing

File: src/Pages/Spotify/index.jsx (hunk 7 above)
Comment (ArgoNavis1763) [reply src/Pages/Spotify/index.jsx]: Moved to env file

File: cypress/e2e/spotify-ticketmaster-API-tests.cy.js (hunk 2 above)
Comment (ArgoNavis1763) [reply cypress/e2e/spotify-ticketmaster-API-tests.cy.js]: Done

File: cypress/support/commands.js (hunk 3 above)
Comment (ArgoNavis1763) [reply cypress/support/commands.js]: done

**Pull request merged**